Classes/Common/DocumentList.php

```diff
@@ -316,6 +252,122 @@ protected function getRecord($element)
         return $record;
     }
 
+    /**
+     * It gets SOLR result
+     *
+     * @access private
+     *
+     * @param array $record: for searched document
+     *
+     * @return Result
+     */
+    private function getSolrResult($record) {
+        $fields = Solr::getFields();
+
+        $query = $this->solr->service->createSelect();
+        // Restrict the fields to the required ones
+        $query->setFields($fields['uid'] .',' . $fields['id'] .',' . $fields['toplevel'] .',' . $fields['thumbnail'] .',' . $fields['page']);
+        foreach ($this->solrConfig as $solr_name) {
+            $query->addField($solr_name);
+        }
+        // Set additional query parameters.
+        // Set reasonable limit for safety reasons.
+        // We don't expect to get more than 10.000 hits per UID.
+        $query->setStart(0)->setRows(1000);
+        // Take over existing filter queries.
+        $filterQueries = isset($this->metadata['options']['params']['filterquery']) ? $this->metadata['options']['params']['filterquery'] : [];
+        // Extend filter query to get all documents with the same UID.
+        foreach ($filterQueries as $key => $value) {
+            if (isset($value['query'])) {
+                $filterQuery[$key] = $value['query'] . ' OR ' . $fields['toplevel'] . ':true';
+                $query->addFilterQuery($filterQuery);
```

Comment:
This won't work either. But I found no way to test this. I think, there is another issue in `Search:makeFacetsMenuArray()`. So I can't get the available facets from Solr and show the facets menu.

Reply:
I have taken look to this method. Do you think that it needs to be adjusted now or it can be done later?

Reply:
The addFilterQuery() above has to be fixed. But before doing this, I'm looking for working facets. Currently, the Solr has no facets available and I still don't know, why.

A query like 

`/solr/dlfCore0/select?facet=on&q=*%3A*&rows=0`

Should return something like

<pre>
{
  "responseHeader":{
    "status":0,
    "QTime":2,
    "params":{
      "q":"*:*",
      "rows":"0",
      "facet":"on"}},
  "response":{"numFound":4517,"start":0,"numFoundExact":true,"docs":[]
  },
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_ranges":{},
    "facet_intervals":{},
    "facet_heatmaps":{}}}
</pre>

In my case - with this PR working - I get

<pre>
{
  "responseHeader":{
    "status":0,
    "QTime":1,
    "params":{
      "q":"*:*",
      "rows":"0",
      "facet":"true"}},
  "response":{"numFound":4376,"start":0,"numFoundExact":true,"docs":[]
  }}
</pre>

That's why no facetting is shown.

Reply:
The faceting is solved (see other commit suggest).

With selecting a facet, we get the (expected) exception:

"A filterquery must have a key value" in Line 283.

The used GET-parameter are:

`tx_dlf[fq][0]=collection_faceting:("Projekt: Illustrierte Magazine der Klassischen Moderne")&tx_dlf[fq][1]=type_faceting:("volume")&tx_dlf[query]=*`

So please rewrite Line 283.